File: toBe/2181B.py
import bisect

def fxn():
    n, m = map(int, input().split(' '))
    a = list(map(int, input().split(' ')))
    b = list(map(int, input().split(' ')))
    a.sort()
    b.sort()
    
    while len(a) > 0 and len(b) > 0:
        temp = b.pop()
        element = a[-1]
        if element >= temp:
            pass
        else:
            bisect.insort(b, temp - element)
        
        if len(b) == 0:
            print("Alice")
            return
        
        temp = a.pop()
        element = b[-1]
        if element >= temp:
            pass
        else:
            bisect.insort(a, temp - element)
        
        if len(a) == 0:
            print("Bob")
            return
        
t: int = int(input())
while t > 0:
    t -= 1
    fxn()

File: toBe/2181B.cpp
#include <bits/stdc++.h>
using namespace std;

void fxn(){
    int n, m;
    cin >> n >> m;

    priority_queue<int> a;
    for(int i = 0; i < n; i++){
        int temp;
        cin >> temp;
        a.push(temp);
    }

    priority_queue<int> b;
    for(int i = 0; i < m; i++){
        int temp;
        cin >> temp;
        b.push(temp);
    }

    int turn = 0;
    while(!a.empty() && !b.empty()){
        if(turn & 1){
            int temp = b.top();
            int temp_b = a.top();
            a.pop();
            if(temp_b - temp > 0){
                a.push(temp_b - temp);
            }
        }
        else {
            int temp = a.top();
            int temp_b = b.top();
            b.pop();
            if(temp_b - temp > 0){
                b.push(temp_b - temp);
            }
        }

        turn++;
    }

    if(a.size() > 0){
        cout << "Alice" << endl;
    }
    else {
        cout << "Bob" << endl;
    }
}
int main(){
    int t;
    cin >> t;
    while(t--){
        fxn();
    }
}